st2common/st2common/models/api/pack.py

```diff
@@ -142,11 +144,24 @@ class PackAPI(BaseAPI):
     }
 
     def __init__(self, **values):
+        name = values.get('name', None)
+
         # Note: If some version values are not explicitly surrounded by quotes they are recognized
         # as numbers so we cast them to string
         if values.get('version', None):
             values['version'] = str(values['version'])
 
+        # Special case for old version which didn't follow semver format (e.g. 0.1, 1.0, etc.)
+        # In case the version doesn't match that format, we simply append ".0" to the end (e.g.
+        # 0.1 -> 0.1.0, 1.0, -> 1.0.0, etc.)
+        if NORMALIZE_PACK_VERSION:
+            new_version = normalize_pack_version(version=values['version'])
+            if new_version != values['version']:
+                LOG.info('Pack "%s" contains invalid semver version specifer, casting it to a full '
```

Comment:
Also, as [noted in Slack](https://stackstorm.slack.com/archives/stackstorm/p1480853691000552),
it would be better with explicit wording and big capslock `DEPRECATION NOTICE`. 

Absolutely clear call to action forcing users to replace it manually.

Eg. if there is no danger, - users more likely to ignore it.